nginx-ingress-istio/README.md

```diff
@@ -0,0 +1,168 @@
+```shell script
+export ISTIO_VERSION=1.4.2; curl -L https://istio.io/downloadIstio | sh -
+```
+## Deploy Istio
+1. Deploy istio components using the [istio operator api](https://istio.io/blog/2019/introducing-istio-operator/)
+Note the usage of `--set` flags to enable mtls and controlPlaneSecurity in the istio mesh to deploy
+```shell script
+./istio-1.4.2/bin/istioctl manifest apply \
+	--set values.global.mtls.enabled=true \
+	--set values.global.controlPlaneSecurityEnabled=true
+```
+2. check statuses of all istio pods and wait for all istio components to be ready
+```shell script
+kubectl get pods -n istio-system
+```
+the expected output should look like:
+```shell script
+NAME                                      READY   STATUS    RESTARTS   AGE
+istio-citadel-6dc789bc4c-chdkb            1/1     Running   0          104s
+istio-galley-78989484fd-przvl             2/2     Running   0          103s
+istio-ingressgateway-7c6c65958-lv8kh      1/1     Running   0          103s
+istio-pilot-66bf767468-5r94m              2/2     Running   0          103s
+istio-policy-84667d9ff8-hhzw7             2/2     Running   1          104s
+istio-sidecar-injector-7d65c79dd5-7dtrg   1/1     Running   0          102s
+istio-telemetry-5dd48bb8bc-24m8d          2/2     Running   2          103s
+prometheus-586d4445c7-22vhb               1/1     Running   0          103s
+```
+
+## Deploy Sample httpbin application
+```shell script
+kubectl label namespace default istio-injection=enabled
+kubectl apply -f ./istio-1.4.2/samples/httpbin/httpbin.yaml
+```
+
+## Set up nginx ingress proxy in the default namespace
+1. Deploy relevant components
+```shell script
+export KUBE_API_SERVER_IP=$(kubectl get svc kubernetes -o jsonpath='{.spec.clusterIP}')/32
+sed "s#__KUBE_API_SERVER_IP__#${KUBE_API_SERVER_IP}#" nginx-default-ns.yaml | kubectl apply -f -
+```
+2. Verify statuses of all pods in the default namespace
+```shell script
+kubectl get pods
+```
+The expected output should look like:
+```shell script
+NAME                                          READY   STATUS    RESTARTS   AGE
+httpbin-64776bf78d-gq4p6                      2/2     Running   0          40s
+nginx-default-http-backend-69cfc8f96b-kszcp   2/2     Running   0          27s
+nginx-ingress-controller-d464877d4-bmgt9      2/2     Running   0          28s
+```
+
+## Set up nginx ingress proxy in the ingress namespace
+1. Deploy nginx-ingress-controller and its default backend
+```shell script
+kubectl create namespace ingress
+kubectl label namespace ingress istio-injection=enabled
+sed "s#__KUBE_API_SERVER_IP__#${KUBE_API_SERVER_IP}#" nginx-ingress-ns.yaml | kubectl apply -f -
+```
+2. Create ingress resource routing to httpbin
+```shell script
+kubectl apply -f ingress-ingress-ns.yaml
+```
+3. Create sidecar resource to allow traffic from pods in ingress namespace to pods in default namespace
+```shell script
+kubectl apply -f sidecar-ingress-ns.yaml
+```
```

Comment:
@ZackButcher Another observation is that as of the current setup, these two steps does not seem to be necessary. The verification steps below behave normally when these steps are omitted